### PR TITLE
Spatiallite3 provider implementation

### LINKED ISSUE #13: sqlite/spatialite provider
Provider based on sqlite/spatialite
- pytest 
- no sqlalchemy (for now?)

### PR DESCRIPTION
**Pull request by jorgejesus**
- Implemented spatiallite3 using generic  SQL
- New dataset using countries from NE
- Sqlite requires install of libsqlite3-mod-spatialite
- Run pyreq to check requirements and versions
- Updated links in README.md to working example the obs was not working
- pytest for sqlite3 provider
- New code checked with flake8

Closes #13

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,11 +33,16 @@ or
 # feature collection metadata
 curl http://localhost:5000/
 # conformance
-curl http://localhost:5000/api/conformance
+curl http://localhost:5000/conformance
 # feature collection
-curl http://localhost:5000/obs
+curl http://localhost:5000/collections/countries
+# feacture colleciion limit 100
```
Comment (tomkralidis) [inline README.md]: s/feacture colleciion/feature collection/

File: README.md [hunk 2]
```diff
@@ -33,11 +33,16 @@ or
...
+curl http://localhost:5000/collections/countries/items?limit=100
 # feature
-curl http://localhost:5000/obs/371
+curl http://localhost:5000/collections/countries/items/1
+# nummer hits
```
Comment (tomkralidis) [inline README.md]: "number of hits"?

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+ENV PYGEOAPI_CONFIG=/pygeoapi/local.yml
+
+WORKDIR /pygeoapi
+RUN pip3 install -r requirements.txt
+RUN pip3 install -r requirements-dev.txt
+RUN pip3 install -e .
+RUN cp pygeoapi-config.yml local.yml
+#export PYGEOAPI_CONFIG=`pwd`/local.yml
+
```
Comment (tomkralidis) [inline docker/Dockerfile]: I think we also have to generate the OpenAPI and set `PYGEOAPI_OPENAPI`, i.e.:

```docker
RUN pygeoapi generate_openapi_document -c $PYGEOAPI_CONFIG > /pygeoapi/openapi.yml
ENV PYGEOAPI_OPENAPI=/pygeoapi/openapi.yml
```

File: pygeoapi/provider/__init__.py [hunk 4]
```diff
@@ -35,7 +35,9 @@
 PROVIDERS = {
     'CSV': 'pygeoapi.provider.csv_.CSVProvider',
     'Elasticsearch': 'pygeoapi.provider.elasticsearch_.ElasticsearchProvider',
-    'GeoJSON': 'pygeoapi.provider.geojson.GeoJSONProvider'
+    'GeoJSON': 'pygeoapi.provider.geojson.GeoJSONProvider',
+    'Sqlite': 'pygeoapi.provider.sqlite.SQLiteProvider',
```
Comment (tomkralidis) [inline pygeoapi/provider/__init__.py]: Should we call this `SQLite` for consistency?

File: tests/test_geojson_provider.py [hunk 5]
```diff
@@ -37,6 +37,7 @@ def config():
 def test_query(fixture, config):
     p = GeoJSONProvider(config)
     results = p.query()
+    print(result)
```
Comment (tomkralidis) [inline tests/test_geojson_provider.py]: remove debugging statement

File: pygeoapi-config.yml [hunk 6]
```diff
@@ -125,3 +125,33 @@ datasets:
...
+            spatial:
+                bbox: [-180,-90,180,90]
+            temporal:
+                begin: None
+                end: now  # or empty
+        provider: 
+            name: Sqlite
+            data: tests/data/ne_110m_admin_0_countries.sqlite
+            id_field: ogc_fid
+            table: ne_110m_admin_0_countries
```
Comment (tomkralidis) [inline pygeoapi-config.yml]: Looks like we have also added `tests/data/ne_110m_lakes.sqlite`? Do we need this SQLite file or can we remove?

Comment (tomkralidis) [reply pygeoapi-config.yml]: As well, we will probably want a README.md in `test/data` with some info on data sources.

Comment (tomkralidis) [reply pygeoapi-config.yml]: Ignore for now, I'll add a README.md in another PR.

File: docker/Dockerfile (hunk 3 above)
Comment (jorgejesus) [reply docker/Dockerfile]: The dockerfile was not supposed to be in this PR will remove it

File: pygeoapi/provider/__init__.py (hunk 4 above)
Comment (jorgejesus) [reply pygeoapi/provider/__init__.py]: Will change it

File: pygeoapi-config.yml (hunk 6 above)
Comment (jorgejesus) [reply pygeoapi-config.yml]: The ne_110m_lakes.sqlite is no  longer needed will remove

File: requirements.txt [hunk 7]
```diff
@@ -1,7 +1,11 @@
-click
-elasticsearch
-flask
-flask_cors
-pyyaml
-requests
-shapely
+elasticsearch==6.2.0
```
Comment (tomkralidis) [inline requirements.txt]: Can we unpin deps? As well:
- move pytest to requirements-dev.txt
- remove setuptools
- do we need pypandoc?
- jinja2 comes with Flask by default; can we remove?

Comment (jorgejesus) [reply requirements.txt]: requirements.txt was autogenerated with pipreqs (https://github.com/bndr/pipreqs)

**Comment by jorgejesus** [comment]
New commit to GeoCat:spatiallite with requested changes

**Pull request merged**